scripts/us_bea/states_gdp/import_data.py

```diff
@@ -0,0 +1,136 @@
+                  'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi',
+                  'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire',
+                  'New Jersey', 'New Mexico', 'New York', 'North Carolina',
+                  'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania',
+                  'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee',
+                  'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
+                  'West Virginia', 'Wisconsin', 'Wyoming']
+    _ZIP_LINK = "https://apps.bea.gov/regional/zip/SQGDP.zip"
+    _STATE_QUARTERLY_GDP_FILE = "SQGDP1__ALL_AREAS_2005_2019.csv"
+
+    def __init__(self):
+        """Downloads the data, cleans it and stores it in instance data frame.
+
+        Returns
+        """
+        df = self.download_data()
+
+        # Filters out columns that are not US states (e.g. New England).
+        df = df[df['GeoName'].isin(self._US_STATES)]
+
+        # Gets columns that represent quarters, e.g. 2015:Q2, by matching
+        # against a regular expression.
+        all_quarters = [q for q in df.columns if re.match(r"....:Q.", q)]
+
+        # Convert table from wide to long format.
+        df = pd.melt(df, id_vars=['GeoFIPS', 'Unit'],
+                     value_vars=all_quarters,
+                     var_name='Quarter')
+
+        qtr_month_map = {
+            'Q1':'03',
+            'Q2':'06',
+            'Q3':'09',
+            'Q4':'12'
+        }
+
+        def date_to_obs_date(date):
+            """Converts date format, e.g., "2005:Q3" to "2005-09"."""
+            return date[:4] + "-" + qtr_month_map[date[5:]]
+        df['Quarter'] = df['Quarter'].apply(date_to_obs_date)
+
+        def convert_geoid(fips_code):
+            """Creates GeoId column. We get lucky that Data Commons's geoIds
+            equal US FIPS state codes.
+            """
+            fips_code = fips_code.replace('"', "")
+            fips_code = fips_code.replace(" ", "")
+            return "geoId/" + fips_code[:2]
+        df['GeoId'] = df['GeoFIPS'].apply(convert_geoid)
+
+        # Set the instance DF to have one row per geoId/quarter pair, with
+        # different measurement methods as columns. This facilitates the
+        # design of TMCFs.
+        self.df = df[df['Unit'] == "Millions of chained 2012 dollars"]
+        self.df.set_index(['GeoId', 'Quarter'])
+        self.df["millions_of_chained_2012_dollars"] = self.df['value']
+        quality_indices = df[df['Unit'] == "Quantity index"]
+        self.df["Quantity_index"] = quality_indices['value'].values
+        current_usd = df[df['Unit'] == "Millions of current dollars"]['value']
+        self.df["millions_of_current_dollars"] = current_usd.values
+        self.df = self.df.drop(["GeoFIPS", "Unit", "value"], axis=1)
+
+    def download_data(self):
+        """Downloads ZIP file, extracts the desired CSV, and puts it into a data
```

Comment:
Could you follow the docstring format for this?

Reply:
Changed this function so that it no longer returns. Now it saves the resulting df to instance variable. Not sure if there is something else that should be specified now that the function doesn't return.